src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -379,7 +380,10 @@ def __init__(
         millisecond: Optional[int] = None,
         microsecond: Optional[int] = None,
         nanosecond: Optional[int] = None,
+        _emit_ast: bool = True,
     ) -> None:
+        from snowflake.snowpark._internal.ast_utils import with_src_position
```

Comment:
Had to import here to prevent circular imports issue.

Reply:
So far in the code, we tried to avoid building expressions directly. The reasoning was that Expressions should disappear as they're not a first-class entity in snowpark. Instead we chose an approach where always the caller (i.e. the function that creates Interval) creates the AST and assigns it to the expression if need be. I.e.,

```
def build_expr_from_snowpark_column_or_python_val(
    expr_builder: proto.Expr, value: ColumnOrLiteral
) -> None:
    """Copy from a Column object's AST, or copy a literal value into an AST expression.

    Args:
        ast (proto.Expr): A previously created Expr() IR entity intance to be filled
        value (ColumnOrLiteral): The value from which to populate the provided ast parameter.

    Raises:
        TypeError: The Expr provided should only be populated from a Snowpark Column with a valid _ast field or a literal value
    """
    if isinstance(value, snowflake.snowpark.Column):
        build_expr_from_snowpark_column(expr_builder, value)
    elif isinstance(value, VALID_PYTHON_TYPES_FOR_LITERAL_VALUE):
        build_expr_from_python_val(expr_builder, value)
    elif isinstance(value, Expression):
        # Expressions must be handled by caller.
        pass
    else:
        raise TypeError(f"{type(value)} is not a valid type for Column or literal AST.")
```

Reply:
we prob. can move the AST logic to build_expr_from_snowpark_column_or_python_val and have an `Interval` case there. I don't think the original intention of snowpark was to use `Interval` directly, but rather the `make_interval` function.